calicoctl/commands/checksystem.go

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	docopt "github.com/docopt/docopt-go"
+	goversion "github.com/mcuadros/go-version"
+)
+
+// The minimum allowed linux kernel version is 2.6.24, which introduced network
+// namespaces and veth pairs.
+const minKernelVersion = "2.6.24"
+
+// Required kernel modules to run Calico
+var requiredModules = []string{"xt_set", "ip6_tables"}
+
+// Checksystem checks host system for compatible versions
```

Comment:
Comment for any exported function in Go has to begin with it's name for godocs to work, that's why it's like `// Checksystem ...`.

Versions being validated are listed in comments (line 31,32)